### PR TITLE
Add overlay for Samsung A40

### PR DESCRIPTION
**Pull request by ponces**

### CODE REVIEW

File: Samsung/A40/res/values/brightness.xml [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+        <item>303</item>
+        <item>319</item>
+        <item>335</item>
+        <item>351</item>
+        <item>367</item>
+        <item>383</item>
+        <item>399</item>
+        <item>415</item>
+        <item>431</item>
+        <item>447</item>
```
Comment (phhusson) [inline Samsung/A40/res/values/brightness.xml]: This won't work properly, AOSP doesn't know anything but 0-255

At the moment, for this device, it's not possible to go higher than 255, though it's easy to add support for going up to 447 (it's in frameworks/base LightsService.java)

My recommendation is to truncate this array to 255

File: Samsung/A40/res/values/notch.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+<dimen name="status_bar_height_portrait">72px</dimen>
+<dimen name="status_bar_height_landscape">72px</dimen>
+<string name="config_mainBuiltInDisplayCutout"/>
```
Comment (phhusson) [inline Samsung/A40/res/values/notch.xml]: Well, Galaxy A40 *does* have display cutout.

Though i'm guessing waterdrop hurts pretty much noone, so we don't care, and it looks like Samsung didn't care as well, else this wouldn't be empty.

@TadiT7 what's you're recommendation there?

File: Samsung/A40/res/values/brightness.xml (hunk 1 above)
Comment (ponces) [reply Samsung/A40/res/values/brightness.xml]: Should the other values be also truncated to 255?
- config_autoBrightnessDisplayValuesNits
- config_screenBrightnessNits
- config_autoBrightnessLevels

**Comment by ponces** [comment]
> I'm happy to see that my overlay generator is arlready put to good use :-)
> 
> Do you have a list of features you know you want to fix with this PR?
> Are they all fixed by this overlay, or are there other issues that might be fixed in an overlay that isn't there?

The purpose was to fix auto brightness on this device as the property didn't appear at all in the Display Settings page...

File: Samsung/A40/res/values/notch.xml (hunk 2 above)
Comment (ponces) [reply Samsung/A40/res/values/notch.xml]: I'm lost here 😁

File: Samsung/A40/res/values/brightness.xml (hunk 1 above)
Comment (phhusson) [reply Samsung/A40/res/values/brightness.xml]: yup, all arrays must have the same size. (there might be some arrays that are size +1, or size -1, check their current size before truncating)

To be clear, when I say "truncate" I mean remove all the values above 255, not replace with 255

Comment (ponces) [reply Samsung/A40/res/values/brightness.xml]: Yeah, I know. Thank you anyway :)
It should be fixed now.

File: Samsung/A40/res/values/notch.xml (hunk 2 above)
Comment (ponces) [reply Samsung/A40/res/values/notch.xml]: This should be fixed with the latest commit (new values retrieved from framework-res__auto_generated_rro.apk)

**Comment by ponces** [comment]
Please, check the latest commit

**Comment by phhusson** [comment]
You've checked auto brightness works with this overlay?

**Comment by ponces** [comment]
> You've checked auto brightness works with this overlay?

I have been testing it during this week and it doesn't seem to be working, even if the option is enabled in the Settings page (adaptive brightness)...
Is there any configuration or debug message that I can search for in a logcat file?

**Comment by phhusson** [comment]
`adb logcat -b all |grep -i brightness -e backlight` on boot should give you some infos I think

**Comment by diggit** [comment]
Hi,
I'd really like to have Adaptive brightness working on A40. Did you give up on it @ponces ?
[Here is gist with grepped log](https://gist.github.com/diggit/151fdf6116d4218beb0311e9e40f0ee4).

**Pull request closed without merging**